fmf/base.py

```diff
@@ -29,10 +33,23 @@
 #  Metadata
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-class Tree:
+class Tree(Mapping[str, Any]):
     """ Metadata Tree """
-
-    def __init__(self, data, name=None, parent=None):
+    parent: Tree | None
+    children: dict[str, Tree]
+    data: dict[str, Any]
```

Comment:
This `dict[str, Any]` tends to appear quite often, representing the "raw" node data. I'd suggest adding an alias for it, so we could tell the difference between "`dict` that holds node data" and "`dict` that is just `dict` with `str`/`Any` content". Let's say `TreeData` or something similar? `data: TreeData` would then be easier to track, and it would make method signatures easier to understand: `child(self, name: str, data: dict[str, Any] | str | None, source: str | None = None) -> None:` vs `child(self, name: str, data: TreeData | None, source: str | None = None) -> None:` where it's immediately obvious what this `data` does actually mean.

Reply:
Good point. Also can it be further refined to only contain json-like types. Maybe:
```
DataType = int | float | str
ListDataType = list[DataType | ListDataType | DictDataType]
DictDataType = dict[str, DataType | ListDataType | DictDataType]
TreeData = dict[str, DataType | ListDataType | DictDataType]
```

Reply:
Yep, definitely possible. Although for the beginning, I for one would be happy with just `TreeData = dict[str, Any]`, exactly because it can be tightened later.